empiric-package/empiric/core/mixins/oracle.py

```diff
@@ -113,6 +114,34 @@ async def get_spot(
             response.num_sources_aggregated,
         )
 
+    async def get_future(
+        self,
+        pair_id,
+        expiry_timestamp,
+        aggregation_mode: AggregationMode = AggregationMode.MEDIAN,
+        # TODO Add sources on the oracle contract and then in the client here
+        # sources=None,
+    ) -> Entry:
```

Comment:
this doesn't return an entry, it returns a tuple

Reply:
would you want this to return a FutureEntry?

Reply:
Good point, it should return a FutureEntry. Then the `get_spot` function also need to be fixed (change tuple to SpotEntry and return type from Entry to SpotEntry)